### DISCUSSION TITLE
Can you use Tab as a trigger to auto complete?

### DISCUSSION THREAD

**Discussion by cricrazy** [espanso/espanso#1167] [Feature requests and Ideas]
For example:

```
  #
  - trigger: "espanso\t"
    replace: "Hi There!"
```

or any other key to invoke Tab key?

Thank you.

**Comment by federico-terzi** [comment]
Hey @cricrazy,

Honestly, I never thought about this possibility, but it can be definitely useful in some situations. 
There are two problems that come to my mind though:
* Some editors replace them with spaces, and therefore espanso can't correctly predict how many characters to delete before applying the match.
* In the majority of text fields, if you press tab after writing, it focuses on the next field, making it impossible to correctly expand the match.

What do you think?

Thanks for your help :)

**Comment by cricrazy** [comment]
@federico-terzi, 

Yes, there are some issues with offering such capability. 

- Text editors tend to have different behavior for the tab functionalities. 
- Also the Command line may behave differently. 

The reason for my suggestions was that many of the snippet completion for Source code editors (like Vim, Notepad++) uses Tab for expanding the code snippet. 

I wonder what other thinks about this?

**Comment by MiloTodt** [comment]
Would using a key other than tab potentially help with the mentioned issues?

**Comment by federico-terzi** [comment]
Hey guys,

> Yes, there are some issues with offering such capability.
> 
> * Text editors tend to have different behavior for the tab functionalities.
> * Also the Command line may behave differently.
> 
> The reason for my suggestions was that many of the snippet completion for Source code editors (like Vim, Notepad++) uses Tab for expanding the code snippet.
> 
> I wonder what other thinks about this?

I'd probably wait to implement TAB support, as it can have many unwanted consequences.

> Would using a key other than tab potentially help with the mentioned issues?

With espanso, you can pretty much use any printable key in the trigger. For example, I usually prepend all my triggers with the `:` character, so that a match becomes `:date`. In the same way, you could use for example a dot at the end, such as `date.`

Let me know what you think

Thanks for your help :)

**Comment by wadbr** [comment]
using ` would produce the same result but without using a modifier key, right? Plus it's very close to the tab key

**Comment by federico-terzi** [comment]
Hey @wadbr,

> using ` would produce the same result but without using a modifier key, right? Plus it's very close to the tab key

Yep, I'd say this could be a great solution to the problem.

Cheers :)

**Comment by freebrowser1** [comment]
This would be very useful. I have been using aText (https://trankynam.com/atext/) , a similar tool for macOS which does support optionally appending a key after the trigger.

Or an option to swallow the latest character of the trigger, e.g.

` - trigger: "foo\t"`
`      replace: bar`
`      swallowlast: true`

which swallows the last character (can be another character instead of a tab).

**Comment by lukeburns** [comment]
Using a space works well enough when you just need a trigger to prevent conflicts, like with LaTeX-style autocompletions: https://gist.github.com/lukeburns/1873f7953443cf6987e92d020fa660ba. Though tab-completion would be nice.

```
  - trigger: "\\c "
    replace: "̧"
  - trigger: "\\cdot "
    replace: "⋅"
```

**Comment by EmSixTeen** [comment]
Using tab to replace the generic ae, oe, and aa with their Nordic equivalents (æ, ø, å) is [how I've always done it with AHK](https://github.com/EmSixTeen/norsk-tab-complete), as that feels very natural when writing. AHK's never had an issue with it, but I'd like to swap this over to Espanso so that I'm not using AHK solely for that.

**Comment by smeech** [comment]
"\t" as a trigger was implemented in v2.2.1 and works fine.
```yml
  - trigger: ":test\t"
    replace: something
```
It was also added to the word separators (#1855) so can be used when `word: true` applies.
